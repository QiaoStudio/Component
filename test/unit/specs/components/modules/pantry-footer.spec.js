import {
  shallowMount
} from 'dp-test'
import ComFooter from 'components/modules/footer/footer.vue'

describe('modules/Com Footer', () => {
  let wrapper
  const links = [
    {
      'columnClass': 'col-lg-2',
      'title': 'GoBear',
      'items': [
        {
          'columnClass': 'col-sm-12',
          'title': 'About us',
          'url': '/sg/about-us'
        }
      ]
    }
  ]
  const socialMedia = {
    'title': 'Follow',
    'items': [
      {
        'icon': 'facebook',
        'url': 'https://www.facebook.com/GoBearSG'
      },
      {
        'icon': 'twitter',
        'url': 'https://twitter.com/xxxxxxsg'
      }
    ]
  }
  const countries = [
    {
      'name': 'hk',
      'url': '/hk'
    }
  ]
  const policyLinks = [
    {
      'title': 'Terms',
      'url': '/sg/terms-and-conditions'
    }
  ]

  function generateWrapper(options) {
    return shallowMount(ComFooter, options)
  }

  afterEach(() => {
    wrapper.destroy()
  })

  it('when footer is set, footer should be shown', () => {
    wrapper = generateWrapper({
      propsData: {
        links: links,
        socialMedia: socialMedia,
        countries: countries,
        copyright: '© 2019 Woodpecker Asia Tech Pte. Ltd. All rights reserved.',
        policyLinks: policyLinks
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('when footer links are set, footer links should be correct', () => {
    wrapper = generateWrapper({
      propsData: {
        links: links
      }
    })
    expect(wrapper.findAll('.container .row:first-child div > span').at(0).text()).toBe('GoBear')
    expect(wrapper.findAll('.container .row:first-child div ul li').length).toBe(1)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('when footer social media is set, footer social media should be correct', () => {
    wrapper = generateWrapper({
      propsData: {
        socialMedia: socialMedia
      }
    })
    expect(wrapper.find('.dp-footer__social > span').text()).toBe('Follow')
    expect(wrapper.findAll('.dp-footer__social ul li').length).toBe(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('when footer countries is set, footer countries should be correct', () => {
    wrapper = generateWrapper({
      propsData: {
        countries: countries
      }
    })
    expect(wrapper.findAll('.dp-footer__country ul li').length).toBe(1)
    expect(wrapper.find('.dp-footer__country ul li a').attributes().href).toBe('/hk')
    expect(wrapper.find('.dp-footer__country ul li a .dp-icon').classes()).toContain('dp-icon__country--hk')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('when footer copyright is set, footer copyright should be correct', () => {
    let copyright = '© 2020 GoBear.'
    wrapper = generateWrapper({
      propsData: {
        copyright: copyright
      }
    })
    expect(wrapper.find('.dp-footer__copyright span').text()).toBe(copyright)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('when footer policyLinks is set, footer policyLinks should be correct', () => {
    wrapper = generateWrapper({
      propsData: {
        policyLinks: policyLinks
      }
    })
    expect(wrapper.findAll('.dp-footer__policy ul li').length).toBe(1)
    expect(wrapper.find('.dp-footer__policy ul li a').attributes().href).toBe('/sg/terms-and-conditions')
    expect(wrapper.find('.dp-footer__policy ul li a').text()).toBe('Terms')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
