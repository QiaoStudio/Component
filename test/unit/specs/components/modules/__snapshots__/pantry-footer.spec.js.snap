// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/Com Footer when footer copyright is set, footer copyright should be correct 1`] = `
<footer class="dp-footer">
  <div class="container">
    <div class="row">
      <!---->
    </div>
    <div class="row">
      <div class="col-xs-10 col-xs-offset-1 col-lg-4 col-lg-offset-0 dp-footer__country">
        <!---->
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__copyright"><span class="dp-footer__copyright">© 2020 GoBear.</span></div>
      <div class="col-xs-12 col-lg-4 dp-footer__policy">
        <!---->
      </div>
    </div>
  </div>
</footer>
`;

exports[`modules/Com Footer when footer countries is set, footer countries should be correct 1`] = `
<footer class="dp-footer">
  <div class="container">
    <div class="row">
      <!---->
    </div>
    <div class="row">
      <div class="col-xs-10 col-xs-offset-1 col-lg-4 col-lg-offset-0 dp-footer__country">
        <ul>
          <li>
            <a href="/hk"><span class="dp-icon dp-icon__country dp-icon__country--hk"></span></a>
          </li>
        </ul>
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__copyright">
        <!---->
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__policy">
        <!---->
      </div>
    </div>
  </div>
</footer>
`;

exports[`modules/Com Footer when footer is set, footer should be shown 1`] = `
<footer class="dp-footer">
  <div class="container">
    <div class="row">
      <div class="col-lg-2"><span class="col-sm-12 dp-footer__title">GoBear</span>
        <ul>
          <li class="col-sm-12">
            <a href="/sg/about-us" class="dp-link">About us</a>
          </li>
        </ul>
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__social"><span class="col-lg-12 dp-footer__title">Follow</span>
        <ul>
          <li>
            <a href="https://www.facebook.com/GoBearSG">
              <i aria-hidden="true" class="fa fa-facebook"></i>
            </a>
          </li>
          <li>
            <a href="https://twitter.com/xxxxxxsg">
              <i aria-hidden="true" class="fa fa-twitter"></i>
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div class="row">
      <div class="col-xs-10 col-xs-offset-1 col-lg-4 col-lg-offset-0 dp-footer__country">
        <ul>
          <li>
            <a href="/hk"><span class="dp-icon dp-icon__country dp-icon__country--hk"></span></a>
          </li>
        </ul>
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__copyright"><span class="dp-footer__copyright">© 2019 Woodpecker Asia Tech Pte. Ltd. All rights reserved.</span></div>
      <div class="col-xs-12 col-lg-4 dp-footer__policy">
        <ul>
          <li>
            <a href="/sg/terms-and-conditions" class="dp-link">Terms</a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</footer>
`;

exports[`modules/Com Footer when footer links are set, footer links should be correct 1`] = `
<footer class="dp-footer">
  <div class="container">
    <div class="row">
      <div class="col-lg-2"><span class="col-sm-12 dp-footer__title">GoBear</span>
        <ul>
          <li class="col-sm-12">
            <a href="/sg/about-us" class="dp-link">About us</a>
          </li>
        </ul>
      </div>
      <!---->
    </div>
    <div class="row">
      <div class="col-xs-10 col-xs-offset-1 col-lg-4 col-lg-offset-0 dp-footer__country">
        <!---->
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__copyright">
        <!---->
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__policy">
        <!---->
      </div>
    </div>
  </div>
</footer>
`;

exports[`modules/Com Footer when footer policyLinks is set, footer policyLinks should be correct 1`] = `
<footer class="dp-footer">
  <div class="container">
    <div class="row">
      <!---->
    </div>
    <div class="row">
      <div class="col-xs-10 col-xs-offset-1 col-lg-4 col-lg-offset-0 dp-footer__country">
        <!---->
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__copyright">
        <!---->
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__policy">
        <ul>
          <li>
            <a href="/sg/terms-and-conditions" class="dp-link">Terms</a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</footer>
`;

exports[`modules/Com Footer when footer social media is set, footer social media should be correct 1`] = `
<footer class="dp-footer">
  <div class="container">
    <div class="row">
      <div class="col-xs-12 col-lg-4 dp-footer__social"><span class="col-lg-12 dp-footer__title">Follow</span>
        <ul>
          <li>
            <a href="https://www.facebook.com/GoBearSG">
              <i aria-hidden="true" class="fa fa-facebook"></i>
            </a>
          </li>
          <li>
            <a href="https://twitter.com/xxxxxxsg">
              <i aria-hidden="true" class="fa fa-twitter"></i>
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div class="row">
      <div class="col-xs-10 col-xs-offset-1 col-lg-4 col-lg-offset-0 dp-footer__country">
        <!---->
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__copyright">
        <!---->
      </div>
      <div class="col-xs-12 col-lg-4 dp-footer__policy">
        <!---->
      </div>
    </div>
  </div>
</footer>
`;
